pkg/metric/writer_cw.go

```diff
@@ -98,11 +116,32 @@ func (w *cloudwatchWriter) WriteOne(ctx context.Context, data *Datum) {
 	w.Write(ctx, Data{data})
 }
 
-func (w *cloudwatchWriter) Write(ctx context.Context, batch Data) {
+func (w *cloudwatchWriter) Write(applicationCtx context.Context, batch Data) {
 	if len(batch) == 0 {
 		return
 	}
 
+	timeout := w.writeTimeout
+	if timeout <= 0 {
+		timeout = defaultWriteTimeout
+	}
+
+	manualCtx, cancel := exec.WithDelayedCancelContext(applicationCtx, timeout)
```

Comment:
There is nothing manual about the context, it should be called `delayedCtx` instead